### ISSUE TITLE
Localization without the dependency on `jupyterlab_server`

### ISSUE THREAD

**Issue by jtpio** [jupyterlite/jupyterlite#349]
### Problem

In #336 we use the optional `jupyterlab_server` dependency to reuse the logic for creating the translation data.

### Proposed Solution

We could instead use `babel` and load data with the `entry_points`:

https://github.com/jupyterlab/language-packs/blob/96819f1765d144f121a05ed66745aba8f80ee276/language-packs/jupyterlab-language-pack-ar-SA/setup.cfg#L22-L24

### Additional context

Follow-up to #336

**Comment by bollwyvl** [comment]
Yeah, dunno... as it's not _entirely_ a standard usage of the `babel` chain, i don't see a huge value to re-implementing it... haven't even done so with Contents API, and this looks _substantially_ more fiddly and WIP-y as more things get translated. Might need a mention in the docs (or we need to make the various addons better self-documenting).

**Comment by jtpio** [comment]
Yeah something to maybe consider later. For now it's fine to rely on `jupyterlab_server` since we already depend on `jupyter_server` for the contents.

For reference this came up in `jupyverse` which uses this approach without `jupyterlab_server`:

https://github.com/jupyter-server/jupyverse/blob/667fc8e0d62099b8f8885f1bec76a4cf785a9a49/plugins/lab/fps_lab/routes.py#L104-L119
